### PR TITLE
GH-34722: [C++][Parquet] Minor: Update wording of Parquet NextPage

### LINKED ISSUE #34722: [C++][Parquet] Minor: Declare lifetime of `shared_ptr<Page>` on PageReader
### Describe the enhancement requested

In C++ Parquet, `SerializedPageReader` holds a `decryption_buffer_` and `decompression_buffer_`, and when `NextPage` called, it will reuse that buffer. So, `Page`'s  buffer data is bounded on `SerializedPageReader::NextPage`, when NextPage is called, it's data might be reset.

I have two problems here:
1. If it's by design, why don't we cache raw data in SerializePageReader?
2. Should we note these in comments?

### Component(s)

C++, Parquet

### PR DESCRIPTION
**Pull request by mapleFU**
### Rationale for this change

This change update wording of Parquet `NextPage`. It will reuse same decompression/decrypt buffer internal. So, use should attension it syntax.

### What changes are included in this PR?

No code change, just add some comments.

### Are these changes tested?

No need

### Are there any user-facing changes?

No.

* Closes: #34722

### CODE REVIEW

**Comment by mapleFU** [comment]
@wjones127 Mind take a look? Just some comments

File: cpp/src/parquet/column_reader.cc [hunk 1]
```diff
@@ -275,6 +275,9 @@ class SerializedPageReader : public PageReader {
   }
 
   // Implement the PageReader interface
+  //
+  // SerializedPageReader reuse same decryption buffer and decompression buffer,
```
Comment (wgtmac) [inline cpp/src/parquet/column_reader.cc:279]:
```suggestion
  // SerializedPageReader reuses same decryption buffer and decompression buffer,
```

Comment (wgtmac) [reply cpp/src/parquet/column_reader.cc:279]: Should we make it clear when and which buffer will be reused? For example,
- If compression is enabled, decompression_buffer_ will be reused.
- If compression is not enabled but encryption is enabled, decryption_buffer_ will be reused.
- Neither encryption nor compression is enabled, the buffer is obtained directly from reading the stream.

File: cpp/src/parquet/column_reader.h [hunk 2]
```diff
@@ -152,6 +152,10 @@ class PARQUET_EXPORT PageReader {
 
   // @returns: shared_ptr<Page>(nullptr) on EOS, std::shared_ptr<Page>
   // containing new Page otherwise
+  //
+  // In SerializedPageReader, it reuses same decryption buffer and
```
Comment (wgtmac) [inline cpp/src/parquet/column_reader.h:156]: This is implementation detail, should we expose it to the doc string?

File: cpp/src/parquet/column_reader.cc (hunk 1 above)
Comment (mapleFU) [reply cpp/src/parquet/column_reader.cc:279]: Seems that `decryption_buffer_` and `decompression_buffer_` is not conflict, and finally returned value might be one of them

File: cpp/src/parquet/column_reader.cc [hunk 3]
```diff
@@ -275,6 +275,10 @@ class SerializedPageReader : public PageReader {
   }
 
   // Implement the PageReader interface
+  //
+  // SerializedPageReader reuses same decryption buffer and decompression buffer
+  // internal, so, if NextPage() is called, content of previous page might be
+  // invalidated.
```
Comment (wjones127) [inline cpp/src/parquet/column_reader.cc:281]:
```suggestion
  // The returned Page contains references that aren't guaranteed to live
  // beyond the next call to NextPage(). SerializedPageReader reuses the 
  // decryption and decompression buffers internally, so if NextPage() is 
  // called then the content of previous page might be invalidated.
```

File: cpp/src/parquet/column_reader.h [hunk 4]
```diff
@@ -152,6 +152,10 @@ class PARQUET_EXPORT PageReader {
 
   // @returns: shared_ptr<Page>(nullptr) on EOS, std::shared_ptr<Page>
   // containing new Page otherwise
+  //
+  // In PageReader, it may reuse same underlying buffer, so, if
+  // NextPage() is called, content of previous page might be
+  // invalidated.
```
Comment (wjones127) [inline cpp/src/parquet/column_reader.h:158]:
```suggestion
  // The returned Page may contain references that aren't guaranteed to live
  // beyond the next call to NextPage().
```

**Review by wjones127 — COMMENTED** [review]
Perhaps we can just frame the API docs around the lifetimes of the returned objects (how long is it guaranteed to be valid?) and not how it is implemented?

**Review by wjones127 — COMMENTED** [review]
Perhaps we can just frame the API docs around the lifetimes of the returned objects (how long is it guaranteed to be valid?) and not how it is implemented?

**Comment by mapleFU** [comment]
Sure, changed it

**Review by wjones127 — APPROVED** [review]

**Pull request merged**